project/Protobuf.scala

```diff
@@ -7,67 +7,71 @@ package lagom
 import sbt._
 import Process._
 import Keys._
-import com.typesafe.sbt.preprocess.Preprocess._
 
 import java.io.File
+import java.io.PrintWriter
 
 object Protobuf {
   val paths = SettingKey[Seq[File]]("protobuf-paths", "The paths that contain *.proto files.")
   val outputPaths = SettingKey[Seq[File]]("protobuf-output-paths", "The paths where to save the generated *.java files.")
+  val importPath = SettingKey[Option[File]]("protobuf-import-path", "The path that contain additional *.proto files that can be imported.")
   val protoc = SettingKey[String]("protobuf-protoc", "The path and name of the protoc executable.")
   val protocVersion = SettingKey[String]("protobuf-protoc-version", "The version of the protoc executable.")
   val generate = TaskKey[Unit]("protobuf-generate", "Compile the protobuf sources and do all processing.")
 
   lazy val settings: Seq[Setting[_]] = Seq(
     paths := Seq((sourceDirectory in Compile).value, (sourceDirectory in Test).value).map(_ / "protobuf"),
     outputPaths := Seq((sourceDirectory in Compile).value, (sourceDirectory in Test).value).map(_ / "java"),
+    importPath := None,
     protoc := "protoc",
     protocVersion := "2.6.1",
     generate := {
       val sourceDirs = paths.value
       val targetDirs = outputPaths.value
+      val log = streams.value.log
 
       if (sourceDirs.size != targetDirs.size)
         sys.error(s"Unbalanced number of paths and destination paths!\nPaths: $sourceDirs\nDestination Paths: $targetDirs")
 
       if (sourceDirs exists (_.exists)) {
         val cmd = protoc.value
-        val log = streams.value.log
+
         checkProtocVersion(cmd, protocVersion.value, log)
 
         val base = baseDirectory.value
         val sources = base / "src"
         val targets = target.value
         val cache = targets / "protoc" / "cache"
 
-        (sourceDirs zip targetDirs) map { case (src, dst) =>
-          val relative = src.relativeTo(sources).getOrElse(throw new Exception(s"path $src is not a in source tree $sources")).toString
-          val tmp = targets / "protoc" / relative
-          IO.delete(tmp)
-          generate(cmd, src, tmp, log)
-          transformDirectory(tmp, dst, _ => true, transformFile(_.replace("com.google.protobuf", "akka.protobuf")), cache, log)
+        (sourceDirs zip targetDirs) map {
+          case (src, dst) ⇒
+            val relative = src.relativeTo(sources).getOrElse(throw new Exception(s"path $src is not a in source tree $sources")).toString
+            val tmp = targets / "protoc" / relative
+            IO.delete(tmp)
+            generate(cmd, src, tmp, log, importPath.value)
+            transformDirectory(tmp, dst, _ ⇒ true, transformFile(_.replace("com.google.protobuf", "akka.protobuf")), cache, log)
```

Comment:
Nitpick:  use `=>` instead of `⇒`. There are others places where the same happens.